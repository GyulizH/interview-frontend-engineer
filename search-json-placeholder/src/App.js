import './App.css';
import AppRoutes from './Routes';
import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchUsers } from './Store/Actions/user';
import { fetchPosts } from './Store/Actions/post';

const App = () => {
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(fetchUsers());
        dispatch(fetchPosts());
    }, [dispatch]);

    return (
        <div className="App">
            <AppRoutes />
        </div>
    );
};

export default App;
