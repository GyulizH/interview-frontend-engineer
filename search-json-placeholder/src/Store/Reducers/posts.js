import { FETCH_POSTS, FETCH_POST } from '../Actions/post';

const initialState = {
    posts: [],
    post: {}
};
export const postReducer = (state = initialState, action) => {
    switch (action.type) {
        case FETCH_POSTS:
            return {
                ...state,
                posts: action.payload
            };
        case FETCH_POST:
            return {
                ...state,
                post: action.payload
            };
        default:
            return state;
    }
};
