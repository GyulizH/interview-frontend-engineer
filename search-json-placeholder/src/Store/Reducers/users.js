import { FETCH_USERS, FETCH_USER } from '../Actions/user';
const initialState = {
    users: [],
    selectedUser: {}
};

export const userReducer = (state = initialState, action) => {
    switch (action.type) {
        case FETCH_USERS:
            return {
                ...state,
                users: action.payload
            };
        case FETCH_USER:
            return {
                ...state,
                selectedUser: action.payload
            };
        default:
            return state;
    }
};
