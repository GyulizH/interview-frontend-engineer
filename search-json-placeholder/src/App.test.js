import { render } from '@testing-library/react';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';
import ReduxThunk from 'redux-thunk';
import { postReducer } from './Store/Reducers/posts';
import { userReducer } from './Store/Reducers/users';
import { App } from './App';

export const createTestStore = () =>
    createStore(
        combineReducers({
            allUsers: userReducer,
            allPosts: postReducer
        }),
        applyMiddleware(ReduxThunk)
    );

describe('App component', () => {
    let store;

    beforeEach(() => {
        store = createTestStore();
    });

    it('should render App component correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <App />
            </Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
