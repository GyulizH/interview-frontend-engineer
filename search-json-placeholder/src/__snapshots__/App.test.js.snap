// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component correctly 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="search-page-container"
    >
      <h3>
        SEARCH USERS AND POSTS
      </h3>
      <div>
        <div
          class="input-container"
        >
          <input
            class="input"
            placeholder="Start typing to search for a user by username or post by title"
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
